app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java

```diff
@@ -0,0 +1,136 @@
+import android.widget.Toast;
+import android.widget.Toolbar;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Random;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import dagger.Binds;
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.auth.SessionManager;
+import fr.free.nrw.commons.theme.NavigationBaseActivity;
+
+/**
+ *  activity for sharing feedback on uploaded activity
+ */
+public class AchievementsActivity extends NavigationBaseActivity {
+
+    private static final double BADGE_IMAGE_WIDTH_RATIO = 0.4;
+    private static final double BADGE_IMAGE_HEIGHT_RATIO = 0.3;
+
+    @BindView(R.id.achievement_badge) ImageView imageView;
+    @BindView(R.id.toolbar) android.support.v7.widget.Toolbar toolbar;
+    @Inject SessionManager sessionManager;
+
+    /**
+     * This method helps in the creation Achievement screen and
+     * dynamically set the size of imageView
+     *
+     * @param savedInstanceState Data bundle
+     */
+    @Override
+    @SuppressLint("StringFormatInvalid")
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_achievements);
+        ButterKnife.bind(this);
+        /**
+         * DisplayMetrics used to fetch the size of the screen
+         */
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
+        int height = displayMetrics.heightPixels;
+        int width = displayMetrics.widthPixels;
+
+        /**
+         * Used for the setting the size of imageView at runtime
+         */
+        RelativeLayout.LayoutParams params = (RelativeLayout.LayoutParams)
+                imageView.getLayoutParams();
+        params.height = (int) (height*BADGE_IMAGE_HEIGHT_RATIO);
+        params.width = (int) (width*BADGE_IMAGE_WIDTH_RATIO);
+        imageView.setImageResource(R.drawable.featured);
+        imageView.requestLayout();
+
+        setSupportActionBar(toolbar);
+        initDrawer();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_about, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+        if (id == R.id.share_app_icon) {
+            View rootView = getWindow().getDecorView().findViewById(android.R.id.content);
+            Bitmap screenShot = getScreenShot(rootView);
+            shareScreen(screenShot);
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    /**
+     * To take screenshot of the screen and return it in Bitmap format
+     * @param view
+     * @return
+     */
+    public static Bitmap getScreenShot(View view) {
```

Comment:
Ideally `getScreenShot` and `shareScreen` methods could move to a util class(maybe `ShareUtils` or something similar). `ShareUtils` should be exposing just one public method to grab a screenshot and then share it. Something like: 

```
ShareUtils.shareScreenshot(view, activity);
// view is the screen to be shared
// activity is the page that initiates it. 
```

This could be potentially reused a lot in the app. The most probable candidate is the **Share Logs** option.